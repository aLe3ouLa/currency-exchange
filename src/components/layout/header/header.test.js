import React from "react";
import ReactDOM from "react-dom";
import Header from "./header.component";
import { cleanup, render } from "@testing-library/react";

import { ThemeProvider } from "styled-components";
import { theme, Defaults } from "../../../styles";

import { BrowserRouter as Router } from "react-router-dom";

describe("Header tests", () => {
  afterEach(cleanup);

  it("renders the Header component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Router>
        <ThemeProvider theme={theme}>
          <React.Fragment>
            <Defaults />
            <Header />
          </React.Fragment>
        </ThemeProvider>
      </Router>,
      div
    );
  });

  it("has a valid snapshot ", () => {
    const { asFragment } = render(
      <Router>
        <ThemeProvider theme={theme}>
          <React.Fragment>
            <Defaults />
            <Header />
          </React.Fragment>
        </ThemeProvider>
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
