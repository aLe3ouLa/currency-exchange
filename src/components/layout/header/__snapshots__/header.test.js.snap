// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests has a valid snapshot  1`] = `
<DocumentFragment>
  <div
    class="sc-AxhCb gvOgAg"
  >
    <div
      class="sc-AxiKw dPWazA"
    >
      <h1
        class="sc-AxjAm hczKRe"
      >
        Dashboard
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
